### PR TITLE
Use pathSegments and Request.change

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: app/lib/analyzer/handlers.dart [hunk 1]
```diff
@@ -16,15 +16,14 @@ import 'models.dart';
...
     '/debug': debugHandler,
-  }[path];
+  }[request.requestedUri.path];
 
   if (handler != null) {
     return handler(request);
-  } else if (path.startsWith('/packages/')) {
-    return packageHandler(request);
+  } else if (request.url.pathSegments.first == 'packages') {
+    return packageHandler(request.change(path: 'packages'));
```
Comment (isoos) [inline app/lib/analyzer/handlers.dart]: This seems to clear the `package/version` part from the path, is that the intention?

Comment (kevmoo) [reply app/lib/analyzer/handlers.dart]: Yup! Let's the handler live it's in sub-path world without caring about the global path mapping

Comment (isoos) [reply app/lib/analyzer/handlers.dart]: If we won't use the `path`, why change it?

Comment (kevmoo) [reply app/lib/analyzer/handlers.dart]: It removes it from the URL – so the handler doesn't have to think about it.

See the code below that changed. You no longer have to ignore the first package segment – it's just not there.

Comment (isoos) [reply app/lib/analyzer/handlers.dart]: I see that below, but I find it confusing that we are changing the path here, as if there is a reason or use for it.

**Comment by kevmoo** [comment]
`path` is weird here. It's not saying "change the path to this" so much as
updating the request.url to be relative to `path` – so it's easier to
reason about the items underneath.

**Comment by kevmoo** [comment]
Not a huge deal, @isoos – can close.

String-based path munging makes me 🤢

**Pull request closed without merging**